### DISCUSSION TITLE
Data won't refresh

### DISCUSSION THREAD

**Discussion by coddipm** [alandtse/tesla#929] [General]
For background, I have a tesla solar panel system with a powerwall 3. I've been using this integration for several months successfully.

Last week I got some messages from Tesla that my gateway device was off line. However, the PW3 doesn't need a separate gateway device. The problem is that my data feeds stopped working at about the time I got the notifications.

Went through the tesla support route and they told me the message were erroneous and to ignore them. They claim they made no changes to the API.

I removed and reinstalled the integration and set it back up. Seems to get the data once but never refreshes. I can get a refresh if I restart HA or reload the integration, but that's it. Any ideas or suggestions on what's going on? Thanks.

**Comment by skipishere** [comment]
Issue #922

The API no longer gives some data and I think at the moment it causes things to stop updating correctly due to an error.
